lib/jobs/redfish-job.js

```diff
@@ -0,0 +1,226 @@
+                    }
+                })
+                .then(self.initClient)
+                .then(function(client) {
+                    return self.collectChassisData(
+                        client, 
+                        data.config.command
+                    );
+                })
+                .then(function(result) {
+                    assert.object(result,
+                        'Expected Result Object For: ' + data.config.command
+                    );
+                    data[data.config.command] = result;
+                    return self._publishRedfishCommandResult(
+                        self.routingKey, 
+                        data.config.command, 
+                        data
+                    );
+                })
+                .then(function() {
+                    return waterline.workitems.findOne({ id: data.workItemId });
+                })
+                .then(function(workitem) {
+                        return waterline.workitems.setSucceeded(null, workitem);
+                })
+                .catch(function (err) {
+                    logger.error("Error Retrieving Redfish Data.", {
+                        data: data,
+                        error: err
+                    });
+                })
+                .finally(function() {
+                    self.removeConcurrentRequest(data.node, data.workItemId);
+                });
+            });
+        })
+        .catch(function(err) {
+            logger.error("Failed to initialize job", { 
+                error:err
+            });
+            self._done(err);
+        });
+    };
+
+    /**
+     * @function concurrentRequests
+     * @description manage concurrent work item command requests
+     */
+    RedfishJob.prototype.concurrentRequests = function(node, workItemId) {
+        assert.string(node);
+        assert.string(workItemId);
+        if(!_.has(this.concurrent, node)){
+            this.concurrent[node] = {};
+        }
+        if(!_.has(this.concurrent[node], workItemId)){
+            this.concurrent[node][workItemId] = 0;
+        }
+        if(this.concurrent[node][workItemId] >= this.maxConcurrent){
+            return true;
+        } else {
+            return false;
+        }
+    };
+
+    /**
+     * @function addConcurrentRequest
+     * @description add a new command request for this work item
+     */
+    RedfishJob.prototype.addConcurrentRequest = function(node, type) {
+        assert.object(this.concurrent[node]);
+        assert.number(this.concurrent[node][type]);
+        this.concurrent[node][type] += 1;
+    };
+
+    /**
+     * @function removeConcurrentRequest
+     * @description remove a completed command request for this work item
+     */
+     RedfishJob.prototype.removeConcurrentRequest = function(node, type) {
+        assert.object(this.concurrent[node]);
+        assert.number(this.concurrent[node][type]);
+        this.concurrent[node][type] -= 1;
```

Comment:
for sake of safety, the minus 1 operation should be done after judging the `concurrent` is bigger than 0